feat(prepare): Update source tracking implementation

aly76 opened the pull request:
Updates to the source tracking implementation for scratch org pools, to align with `force:source:beta` commands. 
Deprecate static resource in scratch orgs for storing source tracking files
Move pool implementation to core library

[Decision Record](https://github.com/Accenture/sfpowerscripts/blob/0aeb28376c72e3e2006030de3676090b19d7dce8/decision%20records/prepare/001-prepare-source-tracking.md)

#817

aly76 committed:
Update SourceTrackingResourceController

Replace soucePathInfos with local source tracking implementation

aly76 committed:
Add source tracking library

azlam-abdulsalam commented on packages/core/src/sfdxwrappers/PushSourceImpl.ts:
```diff
@@ -6,7 +6,7 @@ export default class PushSourceImpl extends SFDXCommand {
 
     public getSFDXCommand(): string {
-        return `sfdx force:source:push`;
+        return `sfdx force:source:beta:push`;
```
What's the idea here?  should we get remove of this wrapper altogether?

aly76 replied:
Will remove beta once GA'd by Salesforce. Also looking to replace with source-deploy-retrieve API in the future.

aly76 commented:
Remaining tasks
- [x] Update SourceTrackingController for SFP CLI
- [X] Investigate whether maxRevision.json can be removed from static resource as well
- [x] Remove static resource
- [x] Remove beta from source commands when are GA

aly76 committed:
Remove static resource

Create maxRevision.json when fetching scratch org, using source tracking library

aly76 committed:
Update source tracking for sfp-cli

azlam-abdulsalam commented:
@aly76  Can we remove duplication between sfp-cli and sfpowerscripts, Lets have a shared library or move to core

aly76 committed:
Remove beta

aly76 committed:
Move pool to core package

aly76 committed:
Merge branch 'develop' into update-source-tracking

aly76 committed:
Merge branch 'develop' into update-source-tracking

azlam-abdulsalam committed:
Merge branch 'develop' into update-source-tracking

azlam-abdulsalam reviewed (approved):
okie
